### ISSUE TITLE
Start JACK client with input ports, but without any input connection ?

### ISSUE THREAD

**Issue by the-drunk-coder** [thestk/rtaudio#60]
Is it possible to start an input/output stream for JACK without connecting to a particular input stream by default ? I always have to set some device, to which the stream is then automatically connected. But i'd basically like to avoid to connect the client to my microphone input by default ... 

I can't figure out any possibility, currently ...

**Comment by garyscavone** [comment]
It’s been a while since I used JACK, but I think RtAudio is simply connecting to the JACK server, with whatever sample rate it is running.  Thus, the connection of devices happens on the JACK server side, not from RtAudio.  Or am I not understanding your question?

**Comment by radarsat1** [comment]
Actually `jack_connect()` is called in RtApiJack::startStream().  https://github.com/thestk/rtaudio/blob/master/RtAudio.cpp#L2460

However, if I remember correctly, whether an application should automatically connect to system output or not is a long-standing discussion in the JACK community.  I don't know if there's currently an accepted standard behaviour.  It's the kind of thing I'd normally expect an environment variable to take care of.. that said, perhaps an option would be nice.  Not sure what the best implementation is.

**Comment by the-drunk-coder** [comment]
Yes, by now i've found the respective lines as well, and added some options to configure this in a customized fork. Works fine for me, with JACK. 

Don't know for other APIs, though.

**Comment by jpcima** [comment]
Having the same JACK problem, I have developed my own solution and proposed it as merge request #68. I then ignored about @the-drunk-coder's version whis is redundant in functionality but implemented slightly differently.
the-drunk-coder/rtaudio@cbb7dcc6605d0d88b600265c9be317dc1915825d
I do not know which code is better suited, in any case I would love to see a patch adopted upstream to solve the JACK issue. Any comments?
